commcare_connect/opportunity/views.py

```diff
@@ -967,11 +953,62 @@ def send_message_mobile_users(request, org_slug=None, pk=None):
     )
 
 
+# used for loading api key dropdown
+@org_member_required
+def get_api_keys(request, org_slug=None):
+    hq_server = request.GET.get("hq_server")
+    if not hq_server:
+        return HttpResponse(
+            format_html("<option value='{}'>{}</option>", None, "Select a HQ Server to load API Keys.")
+        )
+
+    api_keys = HQApiKey.objects.filter(hq_server=hq_server, user=request.user).order_by("-date_created")
+    if not api_keys:
+        return HttpResponse(headers={"HX-Trigger": "no-api-keys-found"})
+
+    options = []
+    options.append(format_html("<option value='{}'>{}</option>", None, "Select an API key"))
+    for api_key in api_keys:
+        api_key_hidden = f"{api_key.api_key[:4]}...{api_key.api_key[-4:]}"
+        options.append(
+            format_html(
+                "<option value='{}'>{}</option>",
+                api_key.id,
+                api_key_hidden,
+            )
+        )
+    return HttpResponse("\n".join(options))
+
+
+# used for loading domain dropdown
+@org_member_required
+def get_domains(request, org_slug=None):
```

Comment:
It's a little weird this view is in the opportunities as well. It should probably be whereever we move the server model.